components/builder-api/src/server/handlers.rs

```diff
@@ -514,7 +525,7 @@ pub fn project_update(req: &mut Request) -> IronResult<Response> {
         (session_token, session_id)
     };
 
-    let (organization, repo) = match req.get::<bodyparser::Struct<ProjectCreateReq>>() {
+    let (organization, repo, token) = match req.get::<bodyparser::Struct<ProjectCreateReq>>() {
```

Comment:
Might want to pull a bunch of this parse code into a central function that can be used by both project_create and project_update

Reply:
good call. I put all of that in `github_creds_from_body` @chefsalim